openeo/rest/job.py

```diff
@@ -25,10 +26,19 @@ class JobLogEntry:
     A Job log entry.
     """
 
-    def __init__(self, log_id: str, level: str, message: str):
-        self.log_id = log_id
-        self.level = level
-        self.message = message
+    def __init__(self, entry):
+        self.log_id = entry['id']
+        self.code = entry['code'] if 'code' in entry else 0
+        self.level = entry['level']
+        self.message = entry['message']
+        self.time =  entry['time'] if 'time' in entry else None # todo: native date/time object?
+        self.path = entry['path'] if 'path' in entry else []
+        self.links = entry['links'] if 'links' in entry else []
+        if 'data' in entry:
+            self.data = entry['data']
+
+    def toJson(self):
+        return json.dumps(self, default = lambda o: o.__dict__) # doesn't work
```

Comment:
This class is not JSON serializable. I followed some blog posts, but couldn't get it working. Tried toJson and toJSON and multiple variants of json.dumps. Could you help, please? @soxofaan

Reply:
what kind of failures do you get?

Reply:
Simply that the class is not serializable. Basically the same as without the method, i.e. adding it did not change anything.

Reply:
I made it work...